packages/dashmate/src/helper/renewZeroSslCertificateHelperFactory.js

```diff
@@ -0,0 +1,62 @@
+const { CronJob } = require('cron');
+
+/**
+ *
+ * @param {listCertificates} listCertificates
+ * @param {renewZeroSSLCertificateTask} renewZeroSSLCertificateTask
+ * @param {DockerCompose} dockerCompose
+ * @return {renewZeroSslCertificateHelper}
+ */
+function renewZeroSslCertificateHelperFactory(
+  listCertificates,
+  renewZeroSSLCertificateTask,
+  dockerCompose,
+) {
+  /**
+   * @typedef renewZeroSslCertificateHelper
+   * @param {Config} config
+   * @return {Promise<void>}
+   */
+  async function renewZeroSslCertificateHelper(config) {
+    const provider = config.get('platform.dapi.envoy.ssl.provider');
+    if (provider !== 'zerossl') {
+      return;
+    }
+
+    const certificatesResponse = await listCertificates(config.get('platform.dapi.envoy.ssl.providerConfigs.zerossl.apiKey'));
+
+    const certificate = certificatesResponse.results.find((result) => result.common_name === config.get('externalIp'));
+
+    if (!certificate) {
+      return;
+    }
+
+    let expiresAt = new Date(certificate.expires);
+    expiresAt.setDay(expiresAt.getDay() - 7);
+
+    if (expiresAt.getTime() < Date.now()) {
+      expiresAt = new Date(Date.now() + 3000);
+    }
+
+    const job = new CronJob(
+      expiresAt, async () => {
+        await renewZeroSSLCertificateTask(config);
+        // restart envoy
+        const serviceInfo = await dockerCompose.inspectService(config.toEnvs(), 'dapi_envoy');
+
+        await dockerCompose.execCommand(config.toEnvs(), serviceInfo.Id, 'kill -SIGHUP 1');
```

Comment:
To make this work, you need to install docker cli (without docker daemon) and mount docker unix socket into this container (`/var/run/docker.sock`)